src/main/java/io/github/noeppi_noeppi/libx/data/provider/SoundDefinitionsProviderBase.java

```diff
@@ -0,0 +1,154 @@
+        this.add(id, definition()
+                .with(sound(id))
+                .subtitle(subtitle));
+    }
+
+    /**
+     * The name for the sound with the amount of different sound files.
+     */
+    protected Sounds sound(String name, int amount) {
+        Sounds sounds = new Sounds();
+        for (int i = 0; i < amount; i++) {
+            sounds.add(sound(this.mod.resource(name + i)));
+        }
+        return sounds;
+    }
+
+    /**
+     * The resource location for the sound with the amount of different sound files.
+     */
+    protected Sounds sound(ResourceLocation name, int amount) {
+        return this.sound(name, SoundDefinition.SoundType.SOUND, amount);
+    }
+
+    protected Sounds sound(ResourceLocation name, SoundDefinition.SoundType type, int amount) {
+        Sounds sounds = new Sounds();
+        for (int i = 0; i < amount; i++) {
+            sounds.add(sound(new ResourceLocation(name.getNamespace(), name.getPath() + i), type));
+        }
+        return sounds;
+    }
+
+    /**
+     * A helper class to allow setting e.g. the volume or pitch for all the sounds in the list.
+     */
+    protected static class Sounds extends ArrayList<SoundDefinition.Sound> {
+
+        public Sounds volume(double volume) {
+            return this.volume((float) volume);
+        }
+
+        public Sounds volume(float volume) {
+            this.iterator().forEachRemaining(sound -> sound.volume(volume));
+            return this;
+        }
+
+        public Sounds pitch(double pitch) {
+            return this.pitch((float) pitch);
+        }
+
+        public Sounds pitch(float pitch) {
+            this.iterator().forEachRemaining(sound -> sound.pitch(pitch));
+            return this;
+        }
+
+        public Sounds weight(int weight) {
+            this.iterator().forEachRemaining(sound -> sound.weight(weight));
+            return this;
+        }
+
+        /**
+         * Same as {@link net.minecraftforge.common.data.SoundDefinition.Sound#stream()} but for all sound files.
+         */
+        public Sounds streamSounds() {
+            return this.stream(true);
+        }
+
+        public Sounds stream(boolean stream) {
+            this.iterator().forEachRemaining(sound -> sound.stream(stream));
+            return this;
+        }
+
+        public Sounds attenuationDistance(int attenuationDistance) {
+            this.iterator().forEachRemaining(sound -> sound.attenuationDistance(attenuationDistance));
+            return this;
+        }
+
+        public Sounds preload() {
+            return this.preload(true);
+        }
+
+        public Sounds preload(boolean preload) {
+            this.iterator().forEachRemaining(sound -> sound.preload(preload));
+            return this;
+        }
+
+        /**
+         * Puts all sounds in one sound definition.
+         */
+        public SoundDefinition asDefinition() {
```

Comment:
Do people need to manually call this? If no, make it `private`. If yes, make it, so it's not required.

Reply:
Store an empty sound definition internally, expose its methods (except `with`) and then after setup, merge the sounds int that definition?